packages/features/calAIPhone/providers/retellAI/services/AgentService.ts

```diff
@@ -498,6 +503,112 @@ export class AgentService {
     };
   }
 
+  async createInboundAgent({
+    name,
+    phoneNumber,
+    userId,
+    teamId,
+    workflowStepId,
+    aiConfigurationService,
+  }: {
+    name?: string;
+    phoneNumber: string;
+    userId: number;
+    teamId?: number;
+    workflowStepId: number;
+    aiConfigurationService: {
+      setupInboundAIConfiguration: () => Promise<{ llmId: string; agentId: string }>;
+    };
+  }) {
+    if (teamId) {
+      const canManage = await this.deps.agentRepository.canManageTeamResources({
+        userId,
+        teamId,
+      });
+      if (!canManage) {
+        throw new HttpError({
+          statusCode: 403,
+          message: "You don't have permission to create agents for this team.",
+        });
+      }
+    }
+
+    const isPhoneNumberValid = isValidPhoneNumber(phoneNumber);
+    if (!isPhoneNumberValid) {
+      throw new HttpError({
+        statusCode: 400,
+        message: "Invalid phone number",
+      });
+    }
+
+    let phoneNumberRecord;
+    if (teamId) {
+      phoneNumberRecord = await this.deps.phoneNumberRepository.findByPhoneNumberAndTeamId({
+        phoneNumber,
+        teamId,
+        userId,
+      });
+    } else {
+      phoneNumberRecord = await this.deps.phoneNumberRepository.findByPhoneNumberAndUserId({
+        phoneNumber,
+        userId,
+      });
+    }
```

Comment:
nit: We could simplify this fn by merging the block under teamId condition with the above teamId condition 

Phone Number validation could be the first thing to be checked, then it makes it easy to merge.